runtime/src/impls.rs

```diff
@@ -49,33 +46,11 @@ pub struct SplitToAllValidators;
 /// The remainder from the division are burned.
 impl OnUnbalanced<NegativeImbalance> for SplitToAllValidators {
 	fn on_nonzero_unbalanced(imbalance: NegativeImbalance) {
-		let validators = Staking::<Runtime>::current_elected();
-		if validators.len().is_zero() || imbalance.peek().is_zero() {
-			return;
-		}
-		// Get a list of elected validators
-		let per_validator_reward: Balance = imbalance.peek() / (validators.len() as Balance);
-
-		// This tracks the total amount of reward actually handed out. Used to adjust total issurance
-		let mut total_imbalance = PositiveImbalance::zero();
-		let asset_id = imbalance.asset_id();
-		for validator in &validators {
-			let dest = Staking::<Runtime>::payee(validator);
-			let reward_destination_account_id = match dest {
-				RewardDestination::Controller => {
-					Staking::<Runtime>::bonded(validator).unwrap_or_else(|| validator.clone())
-				}
-				RewardDestination::Stash | RewardDestination::Staked => validator.clone(),
-			};
-
-			let payout = <GenericAsset<Runtime> as MultiCurrencyAccounting>::deposit_creating(
-				&reward_destination_account_id,
-				Some(asset_id.into()),
-				per_validator_reward,
-			);
-			let _ = total_imbalance.subsume(payout);
+		let amount = imbalance.peek();
+
+		if !amount.is_zero() {
```

Comment:
Is this check neccessary?
I don't think we need to worry if the amount is zero

Reply:
@royyang-centrality actually for me, I'm not sure about this, if we remove the check, there is no problem, just prevent some unnecessary state calculation